File: design-patterns/ts/state/order-status/order-receive-state.ts
// order-receive-state.ts

import OrderContext from "./order-context";
import OrderState from "./order-state";


class OrderReceiveState extends OrderState {
    constructor(context: OrderContext) {
        super(context);
    }

    process(): void {
        // Write code to process the order
        console.log("Order received");

        this.context.setState(null);
    }
}

export default OrderReceiveState;

File: design-patterns/go/state/order_state/order_receive_state.go
// order_receive_state.go

package main

import "fmt"

type OrderReceiveState struct {
	*OrderState
}

func NewOrderReceiveState(context *OrderContext) (orderReceiveState *OrderReceiveState) {
	orderReceiveState = &OrderReceiveState{}
	orderReceiveState.OrderState = NewOrderState(orderReceiveState, context)
	return
}

func (orderReceiveState *OrderReceiveState) Process() {
	fmt.Println("Order received")
	orderReceiveState.context.SetState(nil)
}
